src/toil/lib/resources.py

```diff
@@ -18,6 +18,28 @@
 import resource
 from typing import List, Tuple
 
+# Store some extra usage to tack onto the stats as module-level globals
+_extra_cpu_seconds: float = 0
+_extra_memory_ki: int = 0
+
+def record_extra_memory(peak_ki: int) -> None:
+    """
+    Become responsible for the given peak memory usage, in kibibytes.
+
+    The memory will be treated as if it was used by a child process at the time
+    our real child processes were also using their peak memory.
+    """
+    global _extra_memory_ki
```

Comment:
Is there a way to do this without globals?  We've managed to avoid them so far in Toil.

Reply:
I don't think there's a way to avoid module-level state and still have these functions affect the operation of the other functions, with the functions where they are.

I could hide the state as an attribute of the function objects, or in a list that I can write to elements of, and avoid the `global` keyword.

Or I could introduce a class with some class-level attributes and put the state or the state and all the functions in there. Maybe making everybody static methods on a `ResourceMonitor` class would be better, even though I'd have to fix all the existing usages.